racket/src/cs/core/box.ss

```diff
@@ -2,11 +2,15 @@
   (check who mutable-box? :contract "(and/c box? (not/c immutable?))" b)
   (unsafe-box*-cas! b v1 v2))
 
+(define global-lock (make-mutex))
+
 (define (unsafe-box*-cas! b v1 v2)
-  (with-interrupts-disabled
-   (and (eq? v1 (#3%unbox b))
-        (#3%set-box! b v2)
-        #t)))
+  (mutex-acquire  global-lock)
+  (begin0 (with-interrupts-disabled
```

Comment:
To make the style more consistent, please use `begin0` only on its own line. (Picky, but being more picky about formatting is one of many things I'm trying out with racket7.) Note also the extra space in the previous line.